README.md
<img src="https://raw.githubusercontent.com/xyhxx/program_preview/master/logo/dialog.png" />

English | <a href="https://github.com/xyhxx/proste_dialog/blob/master/doc/cn.md" >中文</a>

# Introduce

The function and design idea of package refer to <a href="https://github.com/xsahil03x/giffy_dialog">giffy_dialog</a>,compared with giffy_dialog, provides higher degree of freedom and stronger scalability,you can use various types of prompts preset in the package, or you can set the widget you want to display.Compared to giffy_ Dialog, customization is not limited to images, you can also pass in any widget or custom animation.proste_dialog needs ShowDialog to display. If you want to use animation effects in the display process, you can refer to the <a href="https://github.com/JackJonson/flutter_animated_dialog">flutter_animated_dialog</a>.

# 2.0.0 add new button layout

only set `btnInARow: false` is fine, if you want to adjust the distance between the two buttons or the distance between the button and the border, you can set `confirmButtonMargin`, which only takes effect when `btnsInARow: false`


<img src="https://raw.githubusercontent.com/xyhxx/program_preview/master/proste_dialog/btns%20vertical.png" />


``` dart
ProsteDialog(
  type: _tipType,
  content: Text('this is package preinstall dialog'),
  insetPadding: EdgeInsets.all(15),
  title: Text('this is package preinstall dialog title'),
  titlePadding: EdgeInsets.only(top: 20),
  contentPadding: EdgeInsets.all(15),
  confirmButtonColor: Colors.pink,
  confirmButtonText: Text('confirm', style: TextStyle(color: Colors.white)),
  cancelButtonText: Text('ignore', style: TextStyle(color: Colors.grey[600])),
  showConfirmButton: true,
  showCancelButton: true,
  btnsInARow: false,
  btnPadding: EdgeInsets.symmetric(vertical: 10),
  confirmButtonMargin: EdgeInsets.symmetric(horizontal: 50),
  buttonRadius: 20,
  onConfirm: () {
    print('preinstall confirm pressed');
    Navigator.pop(context);
  },
  onCancel: () {
    print('preinstall cancel pressed');
    Navigator.pop(context);
  },
);
```

# Usage

__Note that if you pass in duration, the dialog will disappear automatically after the specified time. Even if your showconfirmbutton, showcancelbutton and showcloseicon are all true, the button will not be displayed__

__confirmButtonMargin is only butsInARow is false is it valid__

1. `ProsteDialog` dialog of preset animation prompt in package

| parameter | parameterType | default |
| :-- | :---: | :---: |
| insetPadding | EdgeInsets | EdgeInsets.symmetric(horizontal: 40.0, vertical: 24.0) | 
| dialogRadius | double | 5 | 
| type | DialogTipType | DialogTipType.success | 
| title | Widget | null | 
| content | Widget(required) | null |
| titlePadding | EdgeInsets | EdgeInsets.only(top: 15, left: 15, right: 15) | 
| contentPadding | EdgeInsets | EdgeInsets.all(10) |
| confirmButtonText | Text | Text('Confirm') |
| cancelButtonText | Text | Text('Cancel') |
| showConfirmButton | bool | true |
| showCancelButton | bool | true |
| buttonRadius | double | 5 |
| confirmButtonColor | Color | null |
| cancelButtonColor | Color | null |
| onConfirm | void Funtion() | null |
| onCancel | void Funtion() | null |
| duration | Duration | null |
| background | Color | Colors.white |
| elevation | double | 0 |
| shadowColor | Color | null |
| btnInARow | boolean | true |
| btnPadding | EdgeInsets | null |
| confirmButtonMargin | EdgeInsets | null |

<img src="https://raw.githubusercontent.com/xyhxx/program_preview/master/proste_dialog/preview.gif"/>

``` dart 

showDialog(
  context: context,
  builder: (_) => ProsteDialog(
    type: _tipType,
    content: Text('this is package preinstall dialog'),
    insetPadding: EdgeInsets.all(15),
    dialogRadius: 10,
    title: Text('this is package preinstall dialog title'),
    titlePadding: EdgeInsets.only(top: 20),
    contentPadding: EdgeInsets.all(15),
    confirmButtonText: Text('next', style: TextStyle(color: Colors.white)),
    cancelButtonText: Text('back', style: TextStyle(color: Colors.green)),
    showConfirmButton: true,
    showCancelButton: true,
    confirmButtonColor: Colors.cyan,
    cancelButtonColor: Colors.black,
    onConfirm: () {
      print('preinstall confirm pressed');
      Navigator.pop(context);
    },
    onCancel: () {
      print('preinstall cancel pressed');
      Navigator.pop(context);
    },
  ),
);

```

2. `ProsteCustomDialog` customize dialog for header

| parameter | parameterType | default |
| :-- | :---: | :---: |
| insetPadding | EdgeInsets | EdgeInsets.symmetric(horizontal: 40.0, vertical: 24.0) | 
| dialogRadius | double | 5 | 
| header | Widget | null | 
| title | Widget | null | 
| content | Widget(required) | null |
| titlePadding | EdgeInsets | EdgeInsets.only(top: 15, left: 15, right: 15) | 
| contentPadding | EdgeInsets | EdgeInsets.all(10) |
| confirmButtonText | Text | Text('Confirm') |
| cancelButtonText | Text | Text('Cancel') |
| showConfirmButton | bool | true |
| showCancelButton | bool | true |
| buttonRadius | double | 5 |
| confirmButtonColor | Color | null |
| cancelButtonColor | Color | null |
| onConfirm | void Funtion() | null |
| onCancel | void Funtion() | null |
| duration | Duration | null |
| background | Color | Colors.white |
| elevation | double | 0 |
| shadowColor | Color | null |
| btnInARow | boolean | true |
| btnPadding | EdgeInsets | null |
| confirmButtonMargin | EdgeInsets | null |


<img src="https://raw.githubusercontent.com/xyhxx/program_preview/master/proste_dialog/custom2.png" />

``` dart

showDialog(
  context: context,
  builder: (_) => ProsteCustomDialog(
    insetPadding: EdgeInsets.zero,
    dialogRadius: 50,
    header: AnimatedBuilder(
      animation: _animationController,
      builder: (context, child) {
        return Container(
          height: _animation.value,
          alignment: Alignment.center,
          child: CircularProgressIndicator(),
        );
      },
    ),
    title: Text('this is diy title'),
    content: Text('this is diy dialog'),
    titlePadding: EdgeInsets.only(top: 20),
    contentPadding: EdgeInsets.only(top: 50),
    confirmButtonText: Text('next', style: TextStyle(color: Colors.white)),
    cancelButtonText: Text('back', style: TextStyle(color: Colors.green)),
    showConfirmButton: false,
    showCancelButton: true,
    confirmButtonColor: Colors.cyan,
    cancelButtonColor: Colors.black,
    onConfirm: () {
      print('diy confirm pressed');
      Navigator.pop(context);
    },
    onCancel: () {
      print('diy cancel pressed');
      Navigator.pop(context);
    },
  ),
);

```

3. `ProsteAdvertiseDialog` Dialog box with pictures only

| parameter | parameterType | default |
| :-- | :---: | :---: |
| factor | double (0< value <=1) | .9 |
| image | ImageProvider(required) | null |
| closeIconColor | Colors | Colors.white70 |
| onImagePressed | void Function() | null |
| onImagePressed | void Function() | null |
| showCloseIcon | bool | false |
| imageRadius | BorderRadius | BorderRadius.zero |
| closeBtnSize | double | 32 |
| duration | Duration | null |

<img src="https://raw.githubusercontent.com/xyhxx/program_preview/master/proste_dialog/ProsteAdvertiseDialog.png"/>

``` dart

showDialog(
  context: context,
  builder: (_) => ProsteAdvertiseDialog(
    factor: .6,
    showCloseIcon: true,
    closeIconColor: Colors.pink,
    closeBtnSize: 20,
    image: NetworkImage(networkImg),
    imageRadius: BorderRadius.circular(10),
    onImagePressed: () {
      print('image pressed');
      Navigator.pop(context);
    },
    onClosePressed: () {
      print('close pressed');
      Navigator.pop(context);
    },
  ),
);

```
# Issue
If there is any problem or bug, please submit it to <a href="https://github.com/xyhxx/proste_dialog/issues">issue</a>, I will deal with it as soon as I see it, thanks!

CHANGELOG.md
## [2.1.2]

* fix readme

## [2.1.1]

* add logo

## [2.1.0]

* edit title type to Widget

* edit content type to Widget

## [2.0.0]

* add button vertical layout

* add btnPadding

* add confirmButtonMargin, and it is only butsInARow is false is it valid

## [1.2.0]

* add backgroundColor and shadow

## [1.1.0]

* add Key
## [1.0.2]

* add _scaleController.dispose();

## [1.0.1]

* supplement readme

##  [1.0.0]

* release
## [0.0.1-rc] 








test/proste_dialog_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:proste_dialog/proste_dialog.dart';

void main() {
  test('adds one to input values', () {
    final calculator = Calculator();
    expect(calculator.addOne(2), 3);
    expect(calculator.addOne(-7), -6);
    expect(calculator.addOne(0), 1);
    expect(() => calculator.addOne(null), throwsNoSuchMethodError);
  });
}


lib/src/dialog_base.dart
import 'package:flutter/material.dart';

/// 默认间距
const EdgeInsets _defaultInsetPadding =
    EdgeInsets.symmetric(horizontal: 40.0, vertical: 24.0);

class ProsteBaseDialog extends StatelessWidget {
  ProsteBaseDialog({
    Key? key,
    this.insetPadding = _defaultInsetPadding,
    this.dialogRadius = 5,
    this.widget,
    this.title,
    required this.content,
    this.titlePadding = const EdgeInsets.only(top: 15, left: 15, right: 15),
    this.contentPadding = const EdgeInsets.all(10),
    this.confirmButtonText = const Text('Confirm'),
    this.cancelButtonText = const Text('Cancel'),
    this.showConfirmButton = true,
    this.showCancelButton = true,
    this.buttonRadius = 5,
    this.confirmButtonColor,
    this.cancelButtonColor,
    this.onConfirm,
    this.onCancel,
    this.backgroundColor = Colors.white,
    this.elevation = 0,
    this.shadowColor,
    this.btnInARow = true,
    this.btnPadding,
    this.confirmButtonMargin,
  }) : super(key: key);

  /// dialog与屏幕之间的间距
  final EdgeInsets insetPadding;

  /// 对话框的圆角
  final double dialogRadius;

  /// 显示图片
  final Widget? widget;

  /// 标题组件
  final Widget? title;

  /// 标题的padding
  final EdgeInsets titlePadding;

  /// 描述文字
  final Widget content;

  /// 描述的间距
  final EdgeInsets contentPadding;

  /// 确认按钮文字
  final Text confirmButtonText;

  /// 取消按钮文字
  final Text cancelButtonText;

  /// 是否展示确认按钮
  final bool showConfirmButton;

  /// 是否展示取消按钮
  final bool showCancelButton;

  /// 按钮边框圆角
  final double buttonRadius;

  /// 确认按钮背景色
  final Color? confirmButtonColor;

  /// 取消按钮背景色
  final Color? cancelButtonColor;

  /// 确认按钮点击事件
  final void Function()? onConfirm;

  /// 取消按钮点击事件
  final void Function()? onCancel;

  /// dialog背景色
  final Color? backgroundColor;

  /// card的elevation
  final double? elevation;

  /// 阴影颜色
  final Color? shadowColor;

  /// 按钮是否放在同一行
  final bool btnInARow;

  /// 按钮的padding
  final EdgeInsets? btnPadding;

  /// 不是同一行是确认按钮的margin
  final EdgeInsets? confirmButtonMargin;

  /// 标题和描述内容
  List<Widget> _infoWidget() {
    return [
      if (title != null)
        Padding(
          padding: titlePadding,
          child: title,
        ),
      Padding(
        padding: contentPadding,
        child: content,
      ),
    ];
  }

  /// 按钮组
  Widget _btnGroupWidget(BuildContext context) {
    Color themeColor = Theme.of(context).primaryColor;
    return Padding(
      padding: EdgeInsets.only(bottom: 15),
      child: btnInARow
          ? _bntGroupRow(context, themeColor)
          : _btnGroupColulmn(context, themeColor),
    );
  }

  /// 横向按钮组
  Widget _bntGroupRow(BuildContext context, Color color) {
    return Row(
      mainAxisAlignment: showConfirmButton && showCancelButton
          ? MainAxisAlignment.spaceEvenly
          : MainAxisAlignment.center,
      children: [
        if (showCancelButton)
          ElevatedButton(
            style: ButtonStyle(
              backgroundColor: MaterialStateProperty.all(
                  cancelButtonColor ?? Colors.grey[400]),
              shape: MaterialStateProperty.all(
                RoundedRectangleBorder(
                  borderRadius: BorderRadius.circular(buttonRadius),
                ),
              ),
              padding: MaterialStateProperty.all(btnPadding),
            ),
            onPressed: onCancel ?? () => Navigator.pop(context),
            child: Container(child: cancelButtonText),
          ),
        if (showConfirmButton)
          ElevatedButton(
            style: ButtonStyle(
              backgroundColor:
                  MaterialStateProperty.all(confirmButtonColor ?? color),
              shape: MaterialStateProperty.all(
                RoundedRectangleBorder(
                  borderRadius: BorderRadius.circular(buttonRadius),
                ),
              ),
              padding: MaterialStateProperty.all(btnPadding),
            ),
            onPressed: onConfirm ?? () => Navigator.pop(context),
            child: confirmButtonText,
          ),
      ],
    );
  }

  /// 竖向按钮组
  Widget _btnGroupColulmn(BuildContext context, Color color) {
    return Column(
      children: [
        if (showConfirmButton)
          Container(
            width: double.infinity,
            padding:
                confirmButtonMargin ?? EdgeInsets.symmetric(horizontal: 15),
            child: ElevatedButton(
              style: ButtonStyle(
                backgroundColor:
                    MaterialStateProperty.all(confirmButtonColor ?? color),
                shape: MaterialStateProperty.all(
                  RoundedRectangleBorder(
                    borderRadius: BorderRadius.circular(buttonRadius),
                  ),
                ),
                padding: MaterialStateProperty.all(btnPadding),
              ),
              onPressed: onConfirm ?? () => Navigator.pop(context),
              child: confirmButtonText,
            ),
          ),
        if (showCancelButton)
          TextButton(
            style: ButtonStyle(
              overlayColor: MaterialStateProperty.all(Colors.transparent),
              padding: MaterialStateProperty.all(btnPadding),
              tapTargetSize: MaterialTapTargetSize.shrinkWrap,
            ),
            onPressed: onCancel ?? () => Navigator.pop(context),
            child: cancelButtonText,
          ),
      ],
    );
  }

  @override
  Widget build(BuildContext context) {
    final double screenHeight = MediaQuery.of(context).size.height;
    return Dialog(
      elevation: 0,
      backgroundColor: Colors.transparent,
      insetPadding: insetPadding,
      child: OrientationBuilder(
        builder: (context, orientation) {
          bool isPortrait = orientation == Orientation.portrait;
          return ConstrainedBox(
            constraints: BoxConstraints(
              minHeight: 0,
              maxHeight: isPortrait ? screenHeight * .8 : double.infinity,
              minWidth: double.infinity,
              maxWidth: double.infinity,
            ),
            child: Card(
              elevation: elevation,
              shadowColor: shadowColor,
              color: backgroundColor,
              shape: RoundedRectangleBorder(
                borderRadius: BorderRadius.circular(dialogRadius),
              ),
              child: isPortrait
                  ? Column(
                      mainAxisSize: MainAxisSize.min,
                      children: [
                        if (widget != null)
                          Container(
                            clipBehavior: Clip.antiAlias,
                            decoration: BoxDecoration(
                              borderRadius: BorderRadius.only(
                                topLeft: Radius.circular(dialogRadius),
                                topRight: Radius.circular(dialogRadius),
                              ),
                            ),
                            child: widget,
                          ),
                        ..._infoWidget(),
                        if (showConfirmButton || showCancelButton)
                          _btnGroupWidget(context),
                      ],
                    )
                  : Row(
                      mainAxisSize: MainAxisSize.max,
                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
                      crossAxisAlignment: CrossAxisAlignment.stretch,
                      children: [
                        if (widget != null)
                          Expanded(
                            child: Container(
                              clipBehavior: Clip.antiAlias,
                              decoration: BoxDecoration(
                                borderRadius: BorderRadius.only(
                                  topLeft: Radius.circular(dialogRadius),
                                  bottomLeft: Radius.circular(dialogRadius),
                                ),
                              ),
                              child: widget,
                            ),
                          ),
                        Expanded(
                          child: Column(
                            mainAxisAlignment: MainAxisAlignment.center,
                            children: [
                              ..._infoWidget(),
                              if (showConfirmButton || showCancelButton)
                                _btnGroupWidget(context),
                            ],
                          ),
                        ),
                      ],
                    ),
            ),
          );
        },
      ),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:proste_dialog/proste_dialog.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: MyAppHome(),
    );
  }
}

class MyAppHome extends StatefulWidget {
  @override
  _MyAppHomeState createState() => _MyAppHomeState();
}

class _MyAppHomeState extends State<MyAppHome>
    with SingleTickerProviderStateMixin {
  DialogTipType _tipType = DialogTipType.success;
  late Animation _animation;
  late AnimationController _animationController;

  @override
  void initState() {
    super.initState();
    _animationController =
        AnimationController(vsync: this, duration: Duration(seconds: 1));
    _animation =
        Tween<double>(begin: 80, end: 200).animate(_animationController);
  }

  @override
  Widget build(BuildContext context) {
    final String networkImg2 =
        'https://hbimg.huabanimg.com/f9012b3b86d3d90eb3b2fa8742261026c1eaa88b9c44b-CvNWds_fw658/format/webp';
    final String networkImg =
        'https://hbimg.huabanimg.com/06ef627a592f6c6b638720935b5e41782cafafe363c975-d13YKk_fw658/format/webp';
    final String assetImg = 'assets/images/demo_image.jpg';
    return Scaffold(
      appBar: AppBar(),
      body: SingleChildScrollView(
        child: Container(
          alignment: Alignment.center,
          child: Column(
            children: [
              Text('preinstall dialog type'),
              SizedBox(height: 15),
              Row(
                children: [
                  Expanded(
                    child: RadioListTile<DialogTipType>(
                      dense: true,
                      contentPadding: EdgeInsets.zero,
                      value: DialogTipType.success,
                      title: Text('success'),
                      groupValue: _tipType,
                      onChanged: (val) {
                        setState(() {
                          _tipType = val!;
                        });
                      },
                    ),
                  ),
                  Expanded(
                    child: RadioListTile<DialogTipType>(
                      dense: true,
                      contentPadding: EdgeInsets.zero,
                      value: DialogTipType.warn,
                      title: Text('warn'),
                      groupValue: _tipType,
                      onChanged: (val) {
                        setState(() {
                          _tipType = val!;
                        });
                      },
                    ),
                  ),
                  Expanded(
                    child: RadioListTile<DialogTipType>(
                      dense: true,
                      contentPadding: EdgeInsets.zero,
                      value: DialogTipType.error,
                      title: Text('error'),
                      groupValue: _tipType,
                      onChanged: (val) {
                        setState(() {
                          _tipType = val!;
                        });
                      },
                    ),
                  ),
                ],
              ),
              ElevatedButton(
                onPressed: () {
                  showDialog(
                    context: context,
                    builder: (_) => ProsteDialog(
                      onCancel: () {
                        print('text rich close');
                        Navigator.pop(context);
                      },
                      content: Text.rich(
                        TextSpan(children: [
                          TextSpan(text: 'now'),
                          TextSpan(
                              text: 'get', style: TextStyle(color: Colors.red)),
                          TextSpan(
                              text: 'someThings',
                              style: TextStyle(color: Colors.blue))
                        ]),
                      ),
                    ),
                  );
                },
                child: Text('textRich btn'),
              ),
              ElevatedButton(
                onPressed: () {
                  showDialog(
                    context: context,
                    builder: (_) => ProsteCustomDialog(
                      onConfirm: () {
                        print('scroll confirm');
                        Navigator.pop(context);
                      },
                      content: Container(
                        height: 100,
                        child: SingleChildScrollView(
                          child: Column(
                            children: [
                              Text('is is a line'),
                              Text('is is a line'),
                              Text('is is a line'),
                              Text('is is a line'),
                              Text('is is a line'),
                              Text('is is a line'),
                              Text('is is a line'),
                              Text('is is a line'),
                              Text('is is a line'),
                              Text('is is a line'),
                              Text('is is a line'),
                              Text('is is a line'),
                              Text('is is a line'),
                            ],
                          ),
                        ),
                      ),
                      header: Image.asset(
                        assetImg,
                        fit: BoxFit.cover,
                      ),
                    ),
                  );
                },
                child: Text('content scroll dialog'),
              ),
              ElevatedButton(
                onPressed: () {
                  showDialog(
                    context: context,
                    builder: (_) => ProsteDialog(
                      type: _tipType,
                      content: Text('this is package preinstall dialog',
                          style: TextStyle(color: Colors.white)),
                      insetPadding: EdgeInsets.all(15),
                      dialogRadius: 10,
                      title: Text('this is package preinstall dialog title'),
                      titlePadding: EdgeInsets.only(top: 20),
                      contentPadding: EdgeInsets.all(15),
                      confirmButtonText:
                          Text('next', style: TextStyle(color: Colors.white)),
                      cancelButtonText:
                          Text('back', style: TextStyle(color: Colors.green)),
                      showConfirmButton: true,
                      showCancelButton: true,
                      confirmButtonColor: Colors.cyan,
                      cancelButtonColor: Colors.black,
                      backgroundColor: Colors.transparent,
                      elevation: 8,
                      shadowColor: Colors.green,
                      onConfirm: () {
                        print('preinstall confirm pressed');
                        Navigator.pop(context);
                      },
                      onCancel: () {
                        print('preinstall cancel pressed');
                        Navigator.pop(context);
                      },
                      duration: Duration(seconds: 3),
                    ),
                  );
                },
                child: Text('auto close preinstall'),
              ),
              ElevatedButton(
                onPressed: () {
                  showDialog(
                    context: context,
                    builder: (_) => ProsteDialog(
                      type: _tipType,
                      content: Text('this is package preinstall dialog'),
                      insetPadding: EdgeInsets.all(15),
                      dialogRadius: 10,
                      title: Text('this is package preinstall dialog title'),
                      titlePadding: EdgeInsets.only(top: 20),
                      contentPadding: EdgeInsets.all(15),
                      confirmButtonText:
                          Text('next', style: TextStyle(color: Colors.white)),
                      cancelButtonText:
                          Text('back', style: TextStyle(color: Colors.green)),
                      showConfirmButton: true,
                      showCancelButton: true,
                      confirmButtonColor: Colors.cyan,
                      cancelButtonColor: Colors.black,
                      onConfirm: () {
                        print('preinstall confirm pressed');
                        Navigator.pop(context);
                      },
                      onCancel: () {
                        print('preinstall cancel pressed');
                        Navigator.pop(context);
                      },
                    ),
                  );
                },
                child: Text('preinstall dialog'),
              ),
              ElevatedButton(
                onPressed: () {
                  showDialog(
                    context: context,
                    builder: (_) => ProsteDialog(
                      type: _tipType,
                      dialogRadius: 7,
                      content: Text('this is package preinstall dialog'),
                      insetPadding: EdgeInsets.all(15),
                      title: Text('this is package preinstall dialog title'),
                      titlePadding: EdgeInsets.only(top: 20),
                      contentPadding: EdgeInsets.all(15),
                      confirmButtonColor: Colors.pink,
                      confirmButtonText: Text('confirm',
                          style: TextStyle(color: Colors.white)),
                      cancelButtonText: Text('ignore',
                          style: TextStyle(color: Colors.grey[600])),
                      showConfirmButton: true,
                      showCancelButton: true,
                      btnsInARow: false,
                      btnPadding: EdgeInsets.symmetric(vertical: 10),
                      confirmButtonMargin: EdgeInsets.symmetric(horizontal: 50),
                      buttonRadius: 20,
                      onConfirm: () {
                        print('preinstall confirm pressed');
                        Navigator.pop(context);
                      },
                      onCancel: () {
                        print('preinstall cancel pressed');
                        Navigator.pop(context);
                      },
                    ),
                  );
                },
                child: Text('column btn'),
              ),
              ElevatedButton(
                onPressed: () {
                  showDialog(
                    context: context,
                    builder: (_) => ProsteCustomDialog(
                      content: Text('this is auto close dialog'),
                      showCancelButton: true,
                      showConfirmButton: true,
                      dialogRadius: 10,
                      header: Image.asset(
                        assetImg,
                        fit: BoxFit.cover,
                      ),
                      backgroundColor: Colors.pink,
                      duration: Duration(seconds: 2),
                    ),
                  );
                },
                child: Text('auto close dialog'),
              ),
              ElevatedButton(
                onPressed: () {
                  if (_animationController.isCompleted) {
                    _animationController.reverse();
                  } else if (_animationController.isDismissed) {
                    _animationController.forward();
                  }
                  showDialog(
                    context: context,
                    builder: (_) => ProsteCustomDialog(
                      insetPadding: EdgeInsets.symmetric(horizontal: 30),
                      dialogRadius: 8,
                      header: AnimatedBuilder(
                        animation: _animationController,
                        builder: (context, child) {
                          return Container(
                            height: _animation.value,
                            alignment: Alignment.center,
                            child: CircularProgressIndicator(),
                          );
                        },
                      ),
                      title: Text('this is diy title'),
                      content: Text('this is diy dialog'),
                      titlePadding: EdgeInsets.only(top: 20),
                      contentPadding: EdgeInsets.only(top: 50),
                      confirmButtonText:
                          Text('next', style: TextStyle(color: Colors.white)),
                      cancelButtonText:
                          Text('back', style: TextStyle(color: Colors.green)),
                      showConfirmButton: false,
                      showCancelButton: true,
                      confirmButtonColor: Colors.cyan,
                      cancelButtonColor: Colors.black,
                      btnPadding: EdgeInsets.symmetric(horizontal: 100),
                      onConfirm: () {
                        print('diy confirm pressed');
                        Navigator.pop(context);
                      },
                      onCancel: () {
                        print('diy cancel pressed');
                        Navigator.pop(context);
                      },
                    ),
                  );
                },
                child: Text('widget custom dialog'),
              ),
              ElevatedButton(
                onPressed: () {
                  showDialog(
                    context: context,
                    builder: (_) => ProsteCustomDialog(
                      insetPadding: EdgeInsets.all(15),
                      dialogRadius: 15,
                      header: Image.asset(
                        assetImg,
                        fit: BoxFit.cover,
                      ),
                      title: Text('this is title'),
                      content: Text('this is content'),
                      titlePadding: EdgeInsets.only(top: 50),
                      contentPadding: EdgeInsets.symmetric(vertical: 80),
                      confirmButtonText:
                          Text('go', style: TextStyle(color: Colors.white)),
                      cancelButtonText:
                          Text('back', style: TextStyle(color: Colors.green)),
                      showConfirmButton: true,
                      showCancelButton: false,
                      buttonRadius: 80,
                      confirmButtonColor: Colors.pink,
                      cancelButtonColor: Colors.amber,
                      onConfirm: () {
                        print('confirm pressed');
                        Navigator.pop(context);
                      },
                      onCancel: () {
                        print('cancel pressed');
                        Navigator.pop(context);
                      },
                    ),
                  );
                },
                child: Text('image custom dialog'),
              ),
              ElevatedButton(
                onPressed: () {
                  showDialog(
                    context: context,
                    builder: (_) => ProsteAdvertiseDialog(
                      factor: 1,
                      showCloseIcon: true,
                      closeIconColor: Colors.pink,
                      closeBtnSize: 20,
                      image: NetworkImage(networkImg2),
                      imageRadius: BorderRadius.circular(10),
                      onImagePressed: () {
                        print('image pressed');
                        Navigator.pop(context);
                      },
                      onClosePressed: () {
                        print('close pressed');
                        Navigator.pop(context);
                      },
                      duration: Duration(seconds: 2),
                    ),
                  );
                },
                child: Text('auto close adv dialog'),
              ),
              ElevatedButton(
                onPressed: () {
                  showDialog(
                    context: context,
                    builder: (_) => ProsteAdvertiseDialog(
                      factor: .6,
                      showCloseIcon: true,
                      closeIconColor: Colors.pink,
                      closeBtnSize: 20,
                      image: NetworkImage(networkImg),
                      imageRadius: BorderRadius.circular(10),
                      onImagePressed: () {
                        print('image pressed');
                        Navigator.pop(context);
                      },
                      onClosePressed: () {
                        print('close pressed');
                        Navigator.pop(context);
                      },
                    ),
                  );
                },
                child: Text('adv dialog'),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
